src/lib.rs

```diff
@@ -128,8 +139,93 @@ pub struct Line {
     pub points: Vec<Point>,
 }
 
+impl Line {
+    fn segment_length(&self, i: usize) -> Result<f32, &str> {
+        if i + 1 >= self.points.len() {
+            Err("Line segment index out of bounds")
+        } else {
+            Ok(self.points[i].distance(&self.points[i + 1]))
+        }
+    }
 
-#[derive(Default, Debug)]
+    fn length(&self) -> f32 {
+        let mut length = 0.0;
+        for (previous_index, point) in self.points[1..].iter().enumerate() {
+            length += point.distance(&self.points[previous_index]);
+        }
+        length
```

Comment:
```suggestion
        self.points.iter()
            .zip(self.points[1..].iter())
            .map(|(previous_point, point)| previous_point.distance(point))
            .sum()
```

Reply:
Oh, really nice, this `zip()` method. Thanks for guiding me!